app/controllers/users.controller.js

```diff
@@ -52,3 +53,17 @@ exports.signIn = async (req, res, next) => {
     return next(err);
   }
 };
+exports.createAdmin = async (req, res, next) => {
+  const { body } = req;
+  const userData = userSignUpMapper(body);
+  try {
+    const userExists = await userFindByEmail(userData.email);
+    const passwordHash = hash(userData.password);
+    const userAdmin = userExists
+      ? await userUpdate(userExists.id, { role: ADMIN_ROLE })
+      : await userCreate({ ...userData, password: passwordHash, role: ADMIN_ROLE });
+    return res.status(201).send(userSerializer(userAdmin));
```

Comment:
The 201 could be incorrect on update case, so you can do a validation there too.